crates/runtime/src/instance/allocator/pooling/unix.rs

```diff
@@ -58,11 +58,13 @@ pub fn decommit_table_pages(addr: *mut u8, len: usize) -> Result<()> {
     decommit(addr, len, false)
 }
 
+#[cfg(feature = "async")]
 pub fn commit_stack_pages(_addr: *mut u8, _len: usize) -> Result<()> {
     // A no-op as stack pages remain READ|WRITE
     Ok(())
 }
 
+#[cfg(feature = "async")]
```

Comment:
FWIW I've seen a lot of `#[cfg]` annotations being added here like this, but the main purpose of an optional feature in `wasmtime` was to provide the ability to avoid compiling the `wasmtime-fiber` crate since it has custom assembly stubs that don't work everywhere. For a crate like `wasmtime-runtime`, though, including functions which are never used is only a marginal  hit to compile time so I don't think it's necessarily required that we propagate the `async` feature everywhere.